sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -655,6 +655,9 @@ class ResolveSessionCatalog(
   object SessionCatalogAndTable {
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = nameParts match {
       case SessionCatalogAndIdentifier(catalog, ident) =>
+        if (ident.namespace.isEmpty) {
+          throw new AnalysisException(s"Table not found: ${nameParts.quoted}")
```

Comment:
nit: maybe we can check more
```
if (ident.namespace.length != 1) {
  fail("The namespace of tables in the session catalog must have only one name part.")
}
```

Reply:
And shall we move the check into `SessionCatalogAndIdentifier`?

Reply:
If we move the check into `SessionCatalogAndIdentifier`, we need to consolidate the following error messages:
```
global_temp.ns1.ns2.tbl is not a valid TableIdentifier
Unsupported function name 'default.ns1.ns2.fun'
Table or view not found: spark_catalog.t
```
Is that what you wanted?

Reply:
Yea let's unify the error message to "invalid namespace".